slow_conv3d grad_weight: call gemm directly

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66165
* __->__ #65759
* #65758
* #65757
* #65737
* #65726

Differential Revision: [D31257873](https://our.internmc.facebook.com/intern/diff/D31257873)

peterbell10 committed:
slow_conv3d grad_weight: call gemm directly

[ghstack-poisoned]

peterbell10 committed:
Update on "slow_conv3d grad_weight: call gemm directly"




[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257873).

peterbell10 committed:
Update on "slow_conv3d grad_weight: call gemm directly"


Differential Revision: [D31257873](https://our.internmc.facebook.com/intern/diff/D31257873)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257873).

peterbell10 committed:
Update on "slow_conv3d grad_weight: call gemm directly"


Differential Revision: [D31257873](https://our.internmc.facebook.com/intern/diff/D31257873)

[ghstack-poisoned]

peterbell10 committed:
Update on "slow_conv3d grad_weight: call gemm directly"


Differential Revision: [D31257873](https://our.internmc.facebook.com/intern/diff/D31257873)

[ghstack-poisoned]

ngimel commented on aten/src/ATen/native/ConvolutionMM3d.cpp:
```diff
@@ -386,31 +386,33 @@ void slow_conv3d_backward_out_cpu_template(
+  const int64_t ldc = m;
+
+  for (int64_t group = 0; group < groups; ++group) {
+    at::native::cpublas::gemm(
```
same here, we probably need bmm path.

peterbell10 committed:
Update on "slow_conv3d grad_weight: call gemm directly"


Differential Revision: [D31257873](https://our.internmc.facebook.com/intern/diff/D31257873)

[ghstack-poisoned]

peterbell10 committed:
Update on "slow_conv3d grad_weight: call gemm directly"


Differential Revision: [D31257873](https://our.internmc.facebook.com/intern/diff/D31257873)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257873).
